### PR TITLE
Fixes MissingConverterException when receiving data with the rabbitmq input plugin

### PR DESCRIPTION
**Pull request by msvticket**
logstash-plugins/logstash-input-rabbitmq#112 Fixes most cases of MissingConverterException when receiving data with the rabbitmq input plugin.

This is adding support for Byte, Short and (most importantly) Date. But to decrease verbosity I use the functionality in fallbackConvert to decode subclasses of the given classes.
See method ValueReader.readFieldValue in java package amqp-client for which types can appear over AMQP (except that LongString is changed to String by March Hare).

### CODE REVIEW

**Comment by jsvd** [comment]
jenkins test this please

File: logstash-core/src/main/java/org/logstash/Valuefier.java [hunk 1]
```diff
@@ -120,8 +118,7 @@ private static Object fallbackConvert(final Object o, final Class<?> cls) {
         converters.put(
             BigDecimal.class, value -> new RubyBigDecimal(RubyUtil.RUBY, (BigDecimal) value)
         );
-        converters.put(Long.class, LONG_CONVERTER);
-        converters.put(Integer.class, LONG_CONVERTER);
+        converters.put(Number.class, input -> RubyUtil.RUBY.newFixnum(((Number) input).longValue()));
```
Comment (guyboertje) [inline logstash-core/src/main/java/org/logstash/Valuefier.java]: @msvticket 

This is not a good change. The key for the converters map needs to a concrete class. The `o.getClass()` will always be the concrete class so the lookup will not find this Number.class key, forcing the fallback execution.
I think adding `converters.put(Short.class, LONG_CONVERTER);` is all that is needed (and the Date one below) then we have all the Number subclasses covered.

**Comment by msvticket** [comment]
I know that my first commit would mean means fallback execution, but fallbackConvert would only be called once per concrete type. I don't see what the problem with that would be.
Anyway: `converters.put(Short.class, LONG_CONVERTER);`would not be enough; `converters.put(Byte.class, LONG_CONVERTER);` is needed as well.

**Comment by guyboertje** [comment]
The Valuefier class was meticulously crafted by @original-brownbear to eek out the maximum performance as it is called so many times during an Events lifetime. If we can avoid falling back then all the better.

I think we had some discussion about whether Byte in its external usage is a Char in disguise although it might be a conversation I had with myself 😄. In some senses it would be up to the code on the edge where the data comes into logstash to interpret Byte as a character if that is the intent.

**Pull request merged**